### ISSUE TITLE
Operations with empty frame fail

### ISSUE THREAD

**Issue by Garra1980** [modin-project/modin#3428]
When running smth like this 
`python -c "import modin.experimental.pandas as pd, modin.config as cfg; cfg.Engine.put('Ray'); cfg.Backend.put('OmniSci'); df = pd.DataFrame([]); (df + 1).shape"`

exception occurs:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/pandas/base.py", line 3031, in __getattribute__
    attr = super().__getattribute__(item)
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/pandas/dataframe.py", line 329, in shape
    return len(self.index), len(self.columns)
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/pandas/base.py", line 3031, in __getattribute__
    attr = super().__getattribute__(item)
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/pandas/base.py", line 573, in _get_index
    return self._query_compiler.index
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/experimental/backends/omnisci/query_compiler.py", line 446, in _get_index
    return self._modin_frame.index
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/data.py", line 1907, in _get_index
    self._execute()
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/data.py", line 1699, in _execute
    new_table = self._execute_arrow()
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/data.py", line 1785, in _execute_arrow
    return self._op.input[0]._arrow_col_slice(set(self._op.exprs.keys()))
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/data.py", line 1805, in _arrow_col_slice
    table = self._execute_arrow()
  File "/localdisk/izamyati/miniconda3/envs/modintest112/lib/python3.9/site-packages/modin/experimental/engines/omnisci_on_ray/frame/data.py", line 1778, in _execute_arrow
    return pyarrow.Table()
  File "pyarrow/table.pxi", line 1139, in pyarrow.lib.Table.__init__
TypeError: Do not call Table's constructor directly, use one of the `Table.from_*` functions instead.
```

Looks like smth wrong with empty frame

**Comment by Garra1980** [comment]
cc @ienkovich @dchigarev

**Comment by devin-petersohn** [comment]
@Garra1980 is this only with Omnisci?

**Comment by dchigarev** [comment]
Reopening because accessing a column of an empty frame still raises an exception:
```python
import modin.pandas as pd

df = pd.DataFrame(columns=["a", "b"])
print(df["a"]) # KeyError: 'Column F_a does not exist in schema'
```

Output:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/localdisk1/dchigare/repos/modin_bp/modin/pandas/series.py", line 458, in __repr__
    temp_df = self._build_repr_df(num_rows, num_cols)
  File "/localdisk1/dchigare/repos/modin_bp/modin/pandas/base.py", line 152, in _build_repr_df
    if len(self.index) == 0 or (
  File "/localdisk1/dchigare/repos/modin_bp/modin/pandas/base.py", line 3076, in __getattribute__
    attr = super().__getattribute__(item)
  File "/localdisk1/dchigare/repos/modin_bp/modin/pandas/base.py", line 576, in _get_index
    return self._query_compiler.index
  File "/localdisk1/dchigare/repos/modin_bp/modin/experimental/core/storage_formats/omnisci/query_compiler.py", line 423, in _get_index
    return self._modin_frame.index
  File "/localdisk1/dchigare/repos/modin_bp/modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py", line 1854, in _get_index
    self._execute()
  File "/localdisk1/dchigare/repos/modin_bp/modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py", line 1629, in _execute
    new_table = self._execute_arrow()
  File "/localdisk1/dchigare/repos/modin_bp/modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py", line 1717, in _execute_arrow
    return self._op.input[0]._arrow_select(self._op.exprs)
  File "/localdisk1/dchigare/repos/modin_bp/modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py", line 1747, in _arrow_select
    field = table.schema.field(f"F_{expr.column}")
  File "pyarrow/types.pxi", line 1484, in pyarrow.lib.Schema.field
KeyError: 'Column F_a does not exist in schema'
```

**Comment by AndreyPavlenko** [comment]
Should be fixed by #4910

**Comment by AndreyPavlenko** [comment]
Fixed by https://github.com/modin-project/modin/pull/4910
